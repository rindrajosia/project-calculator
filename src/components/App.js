import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';
import '../assets/style.css';

const App = () => {
  const [operate, setOperate] = useState({ total: null, next: null, operation: null });

  const handleClick = buttonName => {
    const calc = calculate({ ...operate }, buttonName);
    setOperate({ ...calc });
  };

  const { next } = { ...operate };
  return (
    <div className="container">
      <Display
        result={next === null ? '0' : next}
      />
      <ButtonPanel clickHandler={handleClick} />
    </div>
  );
};
export default App;
