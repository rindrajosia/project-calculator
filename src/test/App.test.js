import { render } from '@testing-library/react';
import React from 'react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import App from '../components/App';

test('Check if navbar exist', () => {
  render(<App />);
  const navbar = document.querySelector('#navbar');
  expect(navbar).not.toBeNull();
});

test('Click on home link', () => {
  render(<App />);
  const leftClick = { button: 0 };
  userEvent.click(document.querySelector('#home'), leftClick);
  const home = document.querySelector('#home-page');
  expect(home).not.toBeNull();
});

test('Click on calculator link', () => {
  render(<App />);
  const leftClick = { button: 0 };
  userEvent.click(document.querySelector('#calculator'), leftClick);
  const calculator = document.querySelector('#calculator-page');
  expect(calculator).not.toBeNull();
});

test('Click on quote link', () => {
  render(<App />);
  const leftClick = { button: 0 };
  userEvent.click(document.querySelector('#quote'), leftClick);
  const quote = document.querySelector('#quote-page');
  expect(quote).not.toBeNull();
});

it('renders correctly', () => {
  const app = renderer.create(<App />).toJSON();
  expect(app).toMatchSnapshot();
});
