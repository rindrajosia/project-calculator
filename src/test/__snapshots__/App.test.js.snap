// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="menu"
    id="navbar"
  >
    <div>
      Math magicians
    </div>
    <ul
      className="navigation"
    >
      <li>
        <a
          href="/"
          id="home"
          onClick={[Function]}
        >
          Home 
        </a>
      </li>
      <li>
        <a
          href="/Calculator"
          id="calculator"
          onClick={[Function]}
        >
          | Calculator
        </a>
      </li>
      <li>
        <a
          href="/Quote"
          id="quote"
          onClick={[Function]}
        >
          | Quote
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="quote"
    id="quote-page"
  >
    <p>
      Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. â€“William Paul Thurston
    </p>
  </div>,
]
`;
